kafka/src/main/java/com/metamx/tranquility/kafka/writer/TranquilityEventWriter.java

```diff
@@ -0,0 +1,161 @@
+ */
+public class TranquilityEventWriter
+{
+  private static final Logger log = new Logger(TranquilityEventWriter.class);
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+  private static final RetryPolicy RETRY_POLICY = new ExponentialBackoffRetry(1000, 500, 30000);
+
+  private final CuratorFramework curator;
+  private final DataSourceConfig<TranquilityKafkaConfig> dataSourceConfig;
+  private final SimpleTranquilizerAdapter<scala.collection.immutable.Map<String, Object>> simpleTranquilizerAdapter;
+
+  private long rejectedLogCount = 0;
+
+  public TranquilityEventWriter(String topic, DataSourceConfig<TranquilityKafkaConfig> dataSourceConfig)
+  {
+    this.dataSourceConfig = dataSourceConfig;
+
+    final int zkTimeout = Ints.checkedCast(
+        dataSourceConfig.config()
+                        .zookeeperTimeout()
+                        .toStandardDuration()
+                        .getMillis()
+    );
+
+    curator = CuratorFrameworkFactory.builder()
+                                     .connectString(dataSourceConfig.config().zookeeperConnect())
+                                     .connectionTimeoutMs(zkTimeout)
+                                     .retryPolicy(RETRY_POLICY)
+                                     .build();
+
+    final FinagleRegistry finagleRegistry = new FinagleRegistry(
+        FinagleRegistryConfig.builder().build(),
+        new Disco(curator, dataSourceConfig.config())
+    );
+
+    final Tranquilizer<scala.collection.immutable.Map<String, Object>> tranquilizer =
+        ConfigHelper.createTranquilizerWithLocation(
+            dataSourceConfig,
+            finagleRegistry,
+            curator,
+            DruidLocation.create(
+                dataSourceConfig.config().druidIndexingServiceName(),
+                dataSourceConfig.config().useTopicAsDataSource()
+                ? topic
+                : dataSourceConfig.fireDepartment().getDataSchema().getDataSource()
+            )
+        );
+
+    simpleTranquilizerAdapter = SimpleTranquilizerAdapter.wrap(
+        tranquilizer,
+        dataSourceConfig.config().reportDropsAsExceptions()
+    );
+
+    curator.start();
+    simpleTranquilizerAdapter.start();
+  }
+
+  public void send(byte[] message) throws InterruptedException
+  {
+    Map<String, Object> map;
+    try {
+      map = MAPPER.readValue(
+          message, new TypeReference<HashMap<String, Object>>()
+          {
+          }
+      );
+    }
+    catch (IOException e) {
+      if (++rejectedLogCount <= 5
+          || (rejectedLogCount <= 100 && rejectedLogCount % 10 == 0)
+          || rejectedLogCount % 100 == 0) {
+        log.debug(e, "%d message(s) failed to parse as JSON and were rejected", rejectedLogCount);
+      }
+      if (dataSourceConfig.config().reportDropsAsExceptions()) {
+        Throwables.propagate(e);
+      }
+
+      return;
+    }
+
+    simpleTranquilizerAdapter.send(
```

Comment:
This is gonna potentially throw an exception caused by a totally unrelated message (due to how STA works) but I think that's okay, since the consumer is set up to stop all threads and exit if any send throws an exception. A comment pointing out this situation would be good.